compiler-plugin/src/main/java/io/ballerina/stdlib/data/csvdata/compiler/CsvDataTypeValidator.java

```diff
@@ -0,0 +1,396 @@
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.compiler.syntax.tree.TypeDefinitionNode;
+import io.ballerina.compiler.syntax.tree.VariableDeclarationNode;
+import io.ballerina.projects.plugins.AnalysisTask;
+import io.ballerina.projects.plugins.SyntaxNodeAnalysisContext;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import io.ballerina.tools.diagnostics.DiagnosticFactory;
+import io.ballerina.tools.diagnostics.DiagnosticInfo;
+import io.ballerina.tools.diagnostics.DiagnosticSeverity;
+import io.ballerina.tools.diagnostics.Location;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * CsvData Record Field Validator.
+ *
+ * @since 0.1.0
+ */
+public class CsvDataTypeValidator implements AnalysisTask<SyntaxNodeAnalysisContext> {
+
+    private SemanticModel semanticModel;
+    private final HashMap<Location, DiagnosticInfo> allDiagnosticInfo = new HashMap<>();
+    Location currentLocation;
+    private String modulePrefix = Constants.CSVDATA;
+
+    @Override
+    public void perform(SyntaxNodeAnalysisContext ctx) {
+        semanticModel = ctx.semanticModel();
+        List<Diagnostic> diagnostics = semanticModel.diagnostics();
+        boolean erroneousCompilation = diagnostics.stream()
+                .anyMatch(d -> d.diagnosticInfo().severity().equals(DiagnosticSeverity.ERROR));
+        if (erroneousCompilation) {
+            reset();
+            return;
+        }
+
+        ModulePartNode rootNode = (ModulePartNode) ctx.node();
+        updateModulePrefix(rootNode);
+
+        for (ModuleMemberDeclarationNode member : rootNode.members()) {
+            switch (member.kind()) {
+                case FUNCTION_DEFINITION -> processFunctionDefinitionNode((FunctionDefinitionNode) member, ctx);
+                case MODULE_VAR_DECL ->
+                        processModuleVariableDeclarationNode((ModuleVariableDeclarationNode) member, ctx);
+                case TYPE_DEFINITION ->
+                        processTypeDefinitionNode((TypeDefinitionNode) member, ctx);
+            }
+        }
+        reset();
+    }
+
+    private void reset() {
+        semanticModel = null;
+        allDiagnosticInfo.clear();
+        currentLocation = null;
+        modulePrefix = Constants.CSVDATA;
+    }
+
+    private void updateModulePrefix(ModulePartNode rootNode) {
+        for (ImportDeclarationNode importDeclarationNode : rootNode.imports()) {
+            Optional<Symbol> symbol = semanticModel.symbol(importDeclarationNode);
+            if (symbol.isPresent() && symbol.get().kind() == SymbolKind.MODULE) {
+                ModuleSymbol moduleSymbol = (ModuleSymbol) symbol.get();
+                if (isCsvdataImport(moduleSymbol)) {
+                    modulePrefix = moduleSymbol.id().modulePrefix();
+                    break;
+                }
+            }
```

Comment:
Prefer monadic operations with optionals
```suggestion
                 symbol.filter(moduleSymbol -> moduleSymbol.kind() == SymbolKind.MODULE)
                .filter(this::isCsvdataImport)
                .ifPresent(moduleSymbol -> {
                    modulePrefix = moduleSymbol.getName().orElse("");
                });
```